test/test_aqua.jl

```diff
@@ -0,0 +1,14 @@
+module TestAqua
+
+using Aqua
+using ExplicitImports: check_no_implicit_imports, check_no_stale_explicit_imports
+using Test
+using P4est
+
+@testset "Aqua.jl" begin
+    Aqua.test_all(P4est; unbound_args = false)
```

Comment:
Checking for unbounded arguments is turned off because of this function: https://github.com/trixi-framework/P4est.jl/blob/5c6bb24a7020dd4b2ede53ffb89822434969e44a/src/pointerwrappers.jl#L94C1-L97C59. This is because of the two `Union`s. For certain combinations of input types there are unbounded type parameters. We could circumvent this by expanding the function into several ones, but IMHO this doesn't make it nicer. Thus, I propose to just skip the test.